### PR TITLE
Implementation of passenger service

### PR DESCRIPTION
**Pull request by cooperative0zero**
Three layers have been developed: the database access layer, the business logic layer (services) and the controller layer.
Spring Data JPA was used to organize work with the database, Liquibase was used to manage migrations. 
Endpoints for creating new passengers, changing them, and transferring them to the status of deleted (soft delete) have been created.  Also, the passenger service and its database have been containerized using Docker Compose.

### CODE REVIEW

File: src/main/java/com/modsen/software/passenger/controller/PassengerController.java
Comment (psidarovich) [inline src/main/java/com/modsen/software/passenger/controller/PassengerController.java:1]: Good job. Liked your validations

File: src/main/java/com/modsen/software/passenger/dto/PassengerResponse.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class PassengerResponse {
+    private Long id;
+
+    private String fmlNames;
```
Comment (psidarovich) [inline src/main/java/com/modsen/software/passenger/dto/PassengerResponse.java]: Minor: it's hard to understand the purpose of the field without documentation

File: src/main/java/com/modsen/software/passenger/entity/Passenger.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+
+@Entity
+@Table(name = "passengers")
+@Builder
+public class Passenger {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "p_id")
```
Comment (psidarovich) [inline src/main/java/com/modsen/software/passenger/entity/Passenger.java:17]: Why do you name columns in a such way?

File: src/main/java/com/modsen/software/passenger/service/PassengerService.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.modsen.software.passenger.service;
+
+import com.modsen.software.passenger.dto.PassengerResponse;
+import com.modsen.software.passenger.dto.PassengerRequest;
+
+import java.util.List;
+
+public interface PassengerService {
```
Comment (psidarovich) [inline src/main/java/com/modsen/software/passenger/service/PassengerService.java:8]: If you have only one implementation of the interface without aim of creating hierarchy - no need to create interfaces for each service impl

File: src/main/java/com/modsen/software/passenger/service/impl/PassengerServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            throw new PassengerAlreadyExistsException(String.format("Passenger with id = %d already exists", passenger.getId()));
+        } else {
+            passengerRepository.save(passenger);
+        }
+
+        return conversionService.convert(passenger, PassengerResponse.class);
+    }
+
+    @Override
+    @Transactional
```
Comment (psidarovich) [inline src/main/java/com/modsen/software/passenger/service/impl/PassengerServiceImpl.java:61]: Why do you use transactional annotation on this method?

**Review by psidarovich — APPROVED** [review]

File: src/main/java/com/modsen/software/passenger/controller/PassengerController.java
Comment (cooperative0zero) [reply src/main/java/com/modsen/software/passenger/controller/PassengerController.java:1]: Thank you)

File: src/main/java/com/modsen/software/passenger/dto/PassengerResponse.java (hunk 1 above)
Comment (cooperative0zero) [reply src/main/java/com/modsen/software/passenger/dto/PassengerResponse.java]: Oh, I thought this abbreviation was widely used. What should I do here: just write a comment or rename this field to "firstMiddleLastNames"?

File: src/main/java/com/modsen/software/passenger/entity/Passenger.java (hunk 2 above)
Comment (cooperative0zero) [reply src/main/java/com/modsen/software/passenger/entity/Passenger.java:17]: In the book "Relational Databases in examples" (Svyatoslav Kulikov), it was mentioned as one of the requirements for the flexibility of the database structure: "The names of all structures must be mnemonic and follow a single naming convention." This requirement states that specifying prefixes in names is a rather controversial issue and the decision on this is made by the developer. The main thing is to keep consistency in the choices you make.

File: src/main/java/com/modsen/software/passenger/dto/PassengerResponse.java (hunk 1 above)
Comment (psidarovich) [reply src/main/java/com/modsen/software/passenger/dto/PassengerResponse.java]: fullName is a great solution if your string contains first, last and middle names

File: src/main/java/com/modsen/software/passenger/entity/Passenger.java (hunk 2 above)
Comment (psidarovich) [reply src/main/java/com/modsen/software/passenger/entity/Passenger.java:17]: Ok, possible

So many people, so many tastes

File: src/main/java/com/modsen/software/passenger/service/PassengerService.java (hunk 3 above)
Comment (cooperative0zero) [reply src/main/java/com/modsen/software/passenger/service/PassengerService.java:8]: fixed

File: src/main/java/com/modsen/software/passenger/service/impl/PassengerServiceImpl.java (hunk 4 above)
Comment (cooperative0zero) [reply src/main/java/com/modsen/software/passenger/service/impl/PassengerServiceImpl.java:61]: I see there's really no reason to use it.

**Pull request merged**